rosidl_typesupport_fastrtps_cpp/resource/msg__type_support.cpp.em

```diff
@@ -459,6 +462,288 @@ if isinstance(type_, AbstractNestedType):
   return current_alignment - initial_alignment;
 }
 
+@[  if message.structure.has_any_member_with_annotation('key') ]@
+static
+size_t
+_@(message.structure.namespaced_type.name)__max_serialized_key_size(
+  size_t initial_alignment,
+  bool & is_unbounded)
+{
+  size_t current_alignment = initial_alignment;
+  is_unbounded = false;
+@[for member in message.structure.members]@
+@[  if not member.has_annotation('key')]@
+@[  continue]@
+@[  end if]@
+  // Member: @(member.name)
```

Comment:
Let's try to group duplicated code using something similar to [this](https://github.com/ros2/rosidl/blob/rolling/rosidl_generator_cpp/resource/msg__struct.hpp.em#L104)